src/main/kotlin/org/rust/lang/core/psi/ext/RsMacro.kt

```diff
@@ -73,6 +72,26 @@ val RsMacro.macroBodyStubbed: RsMacroBody?
         }
     }
 
+val RsMacro.preferredBraces: MacroBraces
+    get() = stub?.preferredBraces ?: guessPreferredBraces()
+
+/**
+ * Analyses documentation of macro definition to determine what kind of brackets usually used
+ */
+private fun RsMacro.guessPreferredBraces(): MacroBraces {
+    val documentation = documentation()
+    if (documentation.isNullOrEmpty()) return MacroBraces.PARENS
+
+    val macroCallRegex = """(^|[^\p{Alnum}_])$name\s*!\s*(?<brace>[({\[])""".toRegex()
```

Comment:
Let's move it to static b/c regex compilation is heavy stuff

Reply:
I've extracted it into static field
Note, now the regex matches **all** macro calls because we have only one regex (previously, we substituted macro name into it). All irrelevant matches are filtered manually